### PR TITLE
OPSEXP-2119 Activemq 5.17 support on playbook

### PR DESCRIPTION
**Pull request by slohe1**
OPSEXP-2119

### CODE REVIEW

**Review by alxgomz — CHANGES_REQUESTED** [review]
ActiveMQ 5.17 is only supported since acs 7.3.
We need a backward compatible approach

**Comment by alxgomz** [comment]
I would propose to make the activemq versiojn a [role argument](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_reuse_roles.html#role-argument-validation) passed by `acs.yml` and switch log file based on that

File: roles/activemq/meta/argument_specs.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    author: Alfresco
+    options:
+      activemq_version:
+        description: activemq version to use for deployment.
+        required: false
+        default: 5.17.4
+      activemq_log_config:
+        description: activemq log4j and log4j2 config file.
+        required: false
+        default: log4j2
```
Comment (alxgomz) [inline roles/activemq/meta/argument_specs.yml:14]: This should be a role argument only if it allow you to configure activemq to use log4j or log4jv2 regardless of the version but that's not possible.
As this variable is not used anywhere else but in a single task would make it a task variable only, not a role argument.

**Review by alxgomz — CHANGES_REQUESTED** [review]

File: roles/activemq/tasks/main.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@
   ansible.builtin.get_url:
     url: "{{ dependencies_url.activemq }}"
     checksum: sha512:{{ dependencies_url.activemq_sha512_checksum_url }}
-    dest: "{{ download_location }}/apache-activemq-{{ dependencies_version.activemq }}-bin.tar.gz"
+    dest: "{{ download_location }}/apache-activemq-{{ activemq_version }}-bin.tar.gz"
```
Comment (alxgomz) [inline roles/activemq/tasks/main.yml:6]: 👍

File: roles/activemq/tasks/main.yml [hunk 3]
```diff
@@ -100,8 +100,10 @@
       no_log: true
 
     - name: Change location of ActiveMQ logs
+      vars:
+        activemq_log_config: "{{ 'log4j2' if activemq_version >= '5.17.0' else 'log4j' }}"
```
Comment (alxgomz) [inline roles/activemq/tasks/main.yml:104]: Too bad you're still not using the `version` filter mentioned by @gionn . You could end up with unexpected results.
Check the examples https://docs.ansible.com/ansible/latest/collections/ansible/builtin/version_test.html#examples

**Review by alxgomz — CHANGES_REQUESTED** [review]

File: roles/activemq/tasks/main.yml (hunk 3 above)
Comment (slohe1) [reply roles/activemq/tasks/main.yml]: True, I missed that one.

**Review by alxgomz — APPROVED** [review]

File: roles/activemq/tasks/main.yml [hunk 4]
```diff
@@ -100,8 +100,10 @@
       no_log: true
 
     - name: Change location of ActiveMQ logs
+      vars:
+        activemq_log_config: "{% if activemq_version is version('5.17.0', '>=') %}log4j2{% else %}log4j{% endif %}"
       ansible.builtin.replace:
-        path: "{{ activemq_conf }}/log4j.properties"
+        path: "{{ activemq_conf }}/{{ activemq_log_config }}.properties"
         regexp: "activemq.data"
         replace: "activemq.log"
```
Comment (gionn) [inline roles/activemq/tasks/main.yml:108]: not sure what is this task exactly doing but ok 😬

**Review by gionn — CHANGES_REQUESTED** [review]

**Review by gionn — APPROVED** [review]

**Pull request merged**